_posts/2016-11-17-JAX-London-2016.md

```diff
@@ -0,0 +1,691 @@
+
+- Orders and Customers example with Kafka in between as backchannel
+- Both of them connected with the event store
+- Written in Spring
+
+Summary
+
+- Aggregates are the building blocks of microservices
+- Use events to maintain consistency between aggregates
+- Event Sourcing is a good way to implement an event-driven architecture
+
+
+## Secure by design - Eoin Woods
+<img class="float-image p-image" src="{{ '/img/jax-london-2016/eoin-woods.jpg' | prepend: site.baseurl }}" alt="Eoin Wood" style="width: 100px"/>
+
+[Eoin Woods](https://twitter.com/eoinwoodz) went over security and that we should care about it for things like protection against malice, mistakes and mischance, theft, fraud, destruction, disruption,...
+
+He mentioned OWASP top 10 which he approves and should definitely be checked out if you haven't seen it.
+There are four aspects of security practice:
+
+- Secure Application Design
+- Secure Infrastructure Design
+- Secure Application Implementation	
+- Secure infrastructure deployment
+
+He explains that there are many sets of security design principles but he notes that there are many similarities between them at a fundamental level:
+
+- Viege and McGraw (10)
+- OWASP (10)
+- NIST (33)
+- NCSC (europe)(44)
+- Cliff Berg’s set (185)
+
+Out of all these, Eoin has distilled 10 key principles himself:
+
+- Least privilege
+    - Why: Broad privilege allows malicious or accidental access to protected resources
+	- Principle: Limit privileges to the minimum for the context
+    - Tradeoff: It is less convenient, less efficient and more complexity
+    - Example: Run a server process with the minimum privileges
+- Separate responsibilities
+    - Why: Achieve control and accountability, limit the impact of successful attacks, make attacks less attractive
+    - Principle: Separate and compartmentalise responsibilities and privileges
+    - Tradeoff: Development and testing costs, operation complexity, troubleshooting can be more cumbersome
+    - Example: Admins of a submodule should have no access to other module features
+- Trust cautiously
+    - Why: Many security problems are caused by inserting malicious intermediaries in communication paths
+    - Principle: Assume unknown entities are untrusted, have a clear process to establish trust, validate who is connecting
+    - Tradeoff: Operational complexity, reliability, some development overhead
+    - Example: Dont accept untrusted RMI connections, use client certificates, credentials or network controls
+- Simplest solution possible
+    - Why: Security requires understanding of the design, complex design is rarely understood, simplicity allows analysis
+    - Principle: Actively design for simplicity - avoid complex falure modes, implicit behaviour, unnecessary features,…
+    - Tradeoff: Hard decisions on features and sophistication needs serious design effort to be simple
```

Comment:
Hard decisions on features and sophistication needs serious design effort to be simple => Hard decisions on features provided and needs serious design in effort in order to be simple